### PR TITLE
Update Assessment algorithm 

### PR DESCRIPTION
**Pull request by jwarmkessel**
Update Assessment algorithm.

### CODE REVIEW

File: APCAppCore/APCAppCore/Library/APCMotionHistoryReporter.m [hunk 1]
```diff
@@ -144,304 +154,247 @@ -(void)getMotionCoProcessorDataFrom:(NSDate * __nonnull)startDate andEndDate:(NS
...
+            //numberOfDaysFromStartDate provides the difference of days from now to start of task and therefore if there is no difference we are only getting data for one day.
+            numberOfDaysFromStartDate.day += 1;
+
+            NSDateComponents *dateComponent = [[NSDateComponents alloc] init];
+            [dateComponent setDay:-1];
+            NSDate *newStartDate = [[NSCalendar currentCalendar] dateByAddingComponents:dateComponent
+                                                                               toDate:endDate
+                                                                              options:0];
+
+            [weakSelf getMotionCoProcessorDataFrom:newStartDate
```
Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/APCMotionHistoryReporter.m]: Convert to `StrongSelf` before using in block

Comment (YuanZhu-apple) [inline APCAppCore/APCAppCore/Library/APCMotionHistoryReporter.m]: Convert to `StrongSelf` before using in block

**Comment by jwarmkessel** [comment]
@YuanZhu-apple updated

**Comment by YuanZhu-apple** [comment]
r=me.

**Pull request merged**